src/and-cli-restore-azure-storage.ts

```diff
@@ -0,0 +1,165 @@
+    let destinationContainer: string;
+    let destinationSasToken: string;
+    let recursive: boolean = false;
+    let sourceAccount: string;
+    let sourceContainer: string;
+    let sourceSasToken: string;
+
+    // #endregion Variables
+
+    /// -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const restoreAzureStorage = {
+        description() {
+            return "Restore application assets in Azure Storage";
+        },
+        run() {
+            // Check system/command requirements
+            this.validateOrExit();
+
+            /*
+                Following options currently only support syncing utilizing Azure SAS Tokens.
+                Consider future refactor to allow this restore aspect to first generate a SAS
+                Token using a different means of authentication (service provider, active
+                directory, etc...).  Consider using AZ login and a service provider to then
+                generate a SAS before passing that SAS Token into the Azcopy module later instead
+                of forcing the consumer to define a hard coded value or force the consumer to
+                generate one on its own cadence.
+            */
+            const options = {
+                deleteDestination: deleteDestination,
+                destination: {
+                    account: destinationAccount,
+                    container: destinationContainer,
+                    sasToken: destinationSasToken,
+                },
+                recursive: recursive,
+                source: {
+                    account: sourceAccount,
+                    container: sourceContainer,
+                    sasToken: sourceSasToken,
+                },
+            };
+
+            AzureAzcopySync.containers(options);
+
+            Echo.newLine();
+            Echo.success(
+                "Containers successfully synced in Azure Blob Storage!"
+            );
+        },
+        validateOrExit() {
+            const errors = [];
+
+            if (program.deleteDestination != null) {
+                deleteDestination = program.deleteDestination;
+            }
+            if (program.recursive != null) {
+                recursive = program.recursive;
+            }
+
+            // Validate arguments
+            destinationAccount = program.destinationAccount;
+            if (destinationAccount == null) {
+                errors.push("--destination-account is required");
+            }
+
+            destinationContainer = program.destinationContainer;
+            if (destinationContainer == null) {
+                errors.push("--destination-container is required");
+            }
+
+            destinationSasToken = program.destinationSasToken;
+            if (destinationContainer == "" || destinationContainer == null) {
```

Comment:
Looks like this is checking `destinationContainer` instead of `destinationSasToken`.